internal/types/path_test.go

```diff
@@ -220,10 +178,63 @@ func TestRemoveByPath(t *testing.T) {
 			t.Parallel()
 
 			doc := sourceDoc.DeepCopy()
-			doc.RemoveByPath(tc.path...)
+			RemoveByPath(doc, tc.path...)
 			assert.Equal(t, tc.res, doc, tc.name)
 		})
 	}
+
+	t.Run("test array remove by path", func(t *testing.T) {
+		t.Parallel()
+
+		src := must.NotFail(NewArray(
+			"0", float64(42.13), int32(1000), "2",
+			must.NotFail(NewDocument(
+				"document", "abc",
+				"score", float64(42.13),
+				"age", int32(1000),
+				"foo", deepDoc.DeepCopy(),
+			)),
+			must.NotFail(NewArray("1", "2", "3")),
+		))
+
+		type testCase struct {
```

Comment:
We can embed this struct to make it similar to other tests

Reply:
changed